samples/panther/ap/src/mqtt.c

```diff
@@ -0,0 +1,389 @@
+static int try_to_connect(struct mqtt_client_ctx *client_ctx)
+{
+
+	int i = 0;
+
+	while (i++ < APP_CONNECT_TRIES && !client_ctx->mqtt_ctx.connected) {
+		int rc;
+
+		rc = mqtt_tx_connect(&client_ctx->mqtt_ctx,
+				&client_ctx->connect_msg);
+		k_sleep(APP_SLEEP_MSECS);
+		printk("try_to_connect:mqtt_tx_connect return %d connected %d\n",
+				rc, client_ctx->mqtt_ctx.connected);
+		if (rc != 0) {
+			continue;
+		}
+	}
+
+	if (client_ctx->mqtt_ctx.connected) {
+		return 0;
+	}
+
+	return -EINVAL;
+}
+
+int mqtt_pub_init(void *arg)
+{
+	struct mqtt_client_prm *client_prm = (struct mqtt_client_prm *)arg;
+	int rc;
+
+	printk("mqtt_pub_init\n");
+	/* The net_ctx variable must be ready BEFORE passing it to the MQTT API.
+	 */
+	rc = network_setup(&net_ctx, ZEPHYR_ADDR,
+			(const char *)client_prm->server_addr, client_prm->server_port);
+	printk("network_setup return %d\n", rc);
+	if (rc != 0) {
+		goto exit_app;
+	}
+
+	/* Set everything to 0 and later just assign the required fields. */
+	memset(&client_ctx, 0x00, sizeof(client_ctx));
+
+	/* The network context is the only field that must be set BEFORE
+	 * calling the mqtt_init routine.
+	 */
+	client_ctx.mqtt_ctx.net_ctx = net_ctx;
+
+	/* connect, disconnect and malformed may be set to NULL */
+	client_ctx.mqtt_ctx.connect    = connect_cb;
+	client_ctx.mqtt_ctx.disconnect = disconnect_cb;
+	client_ctx.mqtt_ctx.malformed  = malformed_cb;
+
+	client_ctx.mqtt_ctx.net_timeout = APP_TX_RX_TIMEOUT;
+
+	/* Publisher apps TX the MQTT PUBLISH msg */
+	client_ctx.mqtt_ctx.publish_tx = publish_cb;
+
+	rc = mqtt_init(&client_ctx.mqtt_ctx, MQTT_APP_PUBLISHER);
+	printk("mqtt_init return %d\n", rc);
+	if (rc != 0) {
+		goto exit_app;
+	}
+
+	/* The connect message will be sent to the MQTT server (broker).
+	 * If clean_session here is 0, the mqtt_ctx clean_session variable
+	 * will be set to 0 also. Please don't do that, set always to 1.
+	 * Clean session = 0 is not yet supported.
+	 */
+	client_ctx.connect_msg.clean_session = 1;
+	client_ctx.connect_msg.client_id     = client_prm->client_id;
+	client_ctx.connect_msg.client_id_len = strlen(client_prm->client_id);
+
+	client_ctx.connect_msg.keep_alive     = 5;
+
+	client_ctx.connect_msg.user_name      = client_prm->user_id;
+	client_ctx.connect_msg.user_name_len  = strlen(client_prm->user_id);
+	client_ctx.connect_msg.password       = client_prm->password;
+	client_ctx.connect_msg.password_len   = strlen(client_prm->password);
+
+	client_ctx.connect_data    = "CONNECTED";
+	client_ctx.disconnect_data = "DISCONNECTED";
+	client_ctx.publish_data    = "PUBLISH";
+
+	rc = try_to_connect(&client_ctx);
+	printk("mqttInit:try_to_connect return %d\n", rc);
+	if (rc != 0) {
+		goto exit_app;
+	}
+
+	return 0;
+
+exit_app:
+	printk("\nmqttInit:Error!\n");
+	return -1;
+}
+
+/**
+ */
+int mqtt_status_get(void)
+{
+	return mqtt_status;
+}
+
+/**
+ *
+ */
+int mqtt_send(char *topic, char *payload)
+{
+  struct mqtt_publish_msg *pub_msg = &client_ctx.pub_msg;
```

Comment:
Indentation is wrong here.  Please use tabs.  (Add a git commit hook to always pass your changes through checkpatch. The documentation explains how to do it.)